src/NuGet.Core/NuGet.Configuration/PublicAPI.Shipped.txt

```diff
@@ -288,7 +288,6 @@ NuGet.Configuration.SettingsUtility
 NuGet.Configuration.SourceItem
 ~NuGet.Configuration.SourceItem.ProtocolVersion.get -> string
 ~NuGet.Configuration.SourceItem.ProtocolVersion.set -> void
-~NuGet.Configuration.SourceItem.SourceItem(string key, string value, string protocolVersion = "") -> void
```

Comment:
Why not keep this and/or the new overload?: 
`NuGet.Configuration.SourceItem.SourceItem(string key, string value, string protocolVersion = "", string allowInsecureConnections = "")`

Reply:
Thanks for reviewing!
Adding multiple overloads with optional parameters violates a rule in analyzer : 
Symbol 'Saurceltem' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See '[https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md](https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md)' for details.

Reply:
In general, I'd say the solution to analyzer violations for public APIs is to suppress the analyzer for that method, so we don't break backwards compatibility for our package.

However, I expect that the multiple overloads added for the constructor, this specific PR is both API and ABI backwards compatible. It's just that recompiling the same code with different versions of this dll will result in a different method signature being called.

It's probably worthwhile hand editing PublicAPI.Shipped.txt and PublicAPI.Unshipped.txt to keep this 3 string overload in this file, so we know it's an existing API that we should probably keep backwards compat for, not a new API. The Public API analyzer's code fix suggestions are very simplistic, so I find myself often having to manually hand edit these files when it's not the first time I'm adding a new public API.

Reply:
Thanks! I manually edited the PublicAPI files.